adapters/Chartboost/ChartboostAdapter/GADMAdapterChartboostBanner.m

```diff
@@ -0,0 +1,85 @@
+//
+//  GADMAdapterChartboostBanner.m
+//  Adapter
+//
+//  Created by Daniel Barros on 17/07/2019.
+//  Copyright © 2019 Google. All rights reserved.
+//
+
+#import "GADMAdapterChartboostBanner.h"
+
+@interface GADMAdapterChartboostBanner () <CHBBannerDelegate>
+
+@property (nonatomic) NSMapTable<CHBBanner *, id<CHBBannerDelegate>> *bannersToDelegates;
+@property (nonatomic) NSMutableArray<CHBBanner *> *loadingBanners;
+
+@end
+
+@implementation GADMAdapterChartboostBanner
```

Comment:
I believe this still uses a Singleton pattern. The reason why we created a Singleton class for Interstitial and Rewarded is to handle instances because AdMob allows multiple Interstitials/Rewarded Ads instances while Chartboost only allows one (has a singleton pattern).

That said, is it possible to move the Banner implementation to the main adapter class (GADMAdapterChartboost) since Chartboost allows multiple Banner instances? It also removes the need to forward events from this class to the main adapter.

Let us know if there is a specific reason why a Singleton would be preferred for the implementation of Chartboost Banner ads.

Reply:
This makes sense now, I had misunderstood how the GADMAdNetworkAdapter works. Thanks for the explanation!
I moved the banner implementation to the GADMAdapterChartboost class as you suggested.